### ISSUE TITLE
Report CompileTimeErrorCode.SUPER_AS_EXPRESSION in more places

### ISSUE THREAD

**Issue by scheglov** [dart-lang/sdk#43168]
This code currently has no errors in analyzer, but fails with CFE.
I think this is a bug in the analyzer.

```dart
class A {
  int foo() => 0;
}

class B extends A {
  void bar() {
    print(super);
  }
}

main() {
  B().bar();
}
```

**Comment by jonasfj** [comment]
I'm not entirely sure why I was assigned (typo?) :D

**Comment by bwilkerson** [comment]
@jensjoha

**Comment by sgrekhov** [comment]
This is still an issue

```dart
class C {
  foo() {
    print(super); // No error in analyzer, in CFE Error: Can't use 'super' as an expression.
  }
}

main() {
  C().foo();
}
```

`Dart SDK version: 2.19.0-81.0.dev (dev) (Tue Aug 9 14:51:36 2022 -0700) on "windows_x64"`

**Comment by eernstg** [comment]
I agree that this is still an issue in the analyzer, I changed the labels accordingly.

**Comment by bwilkerson** [comment]
> I agree that this is still an issue in the analyzer, I changed the labels accordingly.

I don't understand why this isn't an issue for the parser. My understanding is that `parse(super);` is not valid Dart code because it doesn't conform to the grammar. Such problems should be caught by the parser so that we can avoid exactly the kind of inconsistency between the front end and the analyzer that's been mentioned above.

**Comment by eernstg** [comment]
Oh, it could very well be an issue for the parser, but I just verified that the error is missing for the analyzer, and it is reported for the CFE.
